### PR TITLE
RHEL OVAL deprecation. New errata workflow

### PR DESCRIPTION
**Pull request by javihernandez**
This PR is the web-server side of https://github.com/AlmaLinux/build-system/milestone/7.

Resolves the following issues:
* https://github.com/AlmaLinux/build-system/issues/346
* https://github.com/AlmaLinux/build-system/issues/347
* https://github.com/AlmaLinux/build-system/issues/348
* https://github.com/AlmaLinux/build-system/issues/352
* https://github.com/AlmaLinux/build-system/issues/350
* https://github.com/AlmaLinux/build-system/issues/356

### CODE REVIEW

File: alws/crud/errata.py [hunk 1]
```diff
@@ -156,6 +161,143 @@ async def get_oval_xml(
...
+        .selectinload(models.BuildTaskArtifact.build_task),
+        selectinload(models.NewErrataRecord.references).selectinload(
+            models.NewErrataReference.cve
+        ),
+    )
+
+    platform = await db.execute(
+        select(models.Platform).where(models.Platform.name == platform_name)
+    )
+    platform: models.Platform = platform.scalars().first()
```
Comment (anfimovdm) [inline alws/crud/errata.py:180]: please, make it a subquery instead

File: alws/crud/errata.py [hunk 2]
```diff
@@ -628,6 +768,192 @@ async def get_matching_albs_packages(
...
+    self_ref_exists = False
+    for ref in errata.references:
+        db_cve = None
+        if ref.cve:
+            db_cve = await db.execute(
+                select(models.ErrataCVE).where(
+                    models.ErrataCVE.id == ref.cve.id
+                )
+            )
+            db_cve = db_cve.scalars().first()
```
Comment (anfimovdm) [inline alws/crud/errata.py:785]: we can get these CVE's by one request with something like this:
```python
db_cves = {}
cve_ids = [ref.cve.id for ref in errata.references if ref.cve]
if cve_ids:
  db_cves = {cve.id: cve for cve in await session.execute(...).scalars().all()}
for ref in errata.references:
     ...
    db_cve = db_cves.get(ref)
```

File: alws/crud/errata.py (hunk 1 above)
Comment (javihernandez) [reply alws/crud/errata.py]: done, please check

File: alws/crud/errata.py (hunk 2 above)
Comment (javihernandez) [reply alws/crud/errata.py]: done, please check

File: alws/crud/errata.py [hunk 3]
```diff
@@ -156,6 +161,148 @@ async def get_oval_xml(
...
+                objects.add(obj["id"])
+                oval.append_object(
+                    get_object_cls_by_tag(obj["type"]).from_dict(obj)
+                )
+        for state in record.states:
+            if state["id"] not in objects:
+                objects.add(state["id"])
+                oval.append_object(
+                    get_state_cls_by_tag(state["type"]).from_dict(state)
+                )
```
Comment (anfimovdm) [inline alws/crud/errata.py:294]: I suppose that we can unite these loops into one. 
Something like that:
```python
for coll in (record.tests, record.objects, record.states, record.variables):
    if coll is None:
        continue
    for el in coll:
        if el['id'] in objects:
            continue
        objects.add(el["id"])
        oval.append_object(
            get_state_cls_by_tag(el["type"]).from_dict(el)
        )
```
but, it's up to you

**Review by anfimovdm — APPROVED** [review]

File: alws/crud/errata.py (hunk 3 above)
Comment (javihernandez) [reply alws/crud/errata.py:305]: Yes, I see your point and I actually considered it when implementing it, but every oval ref has its own `get_*_cls_by_tag`, hence I decided to keep it as different loops

**Review by soksanichenko — APPROVED** [review]
LGTM

File: alws/crud/errata.py (hunk 3 above)
Comment (anfimovdm) [reply alws/crud/errata.py:305]: oh, I see that now, but still, we can add these functions to our tuple like that:
```python
for coll, func in (
    (record.tests, get_test_cls_by_tag),
):
    ...
    for el in coll:
        oval.append_object(func(...).from_dict(el))
```

Comment (javihernandez) [reply alws/crud/errata.py]: Ok, turned oval objects addition into a function and followed your loop nesting approach in https://github.com/AlmaLinux/albs-web-server/pull/998/commits/d36ac66ce566bd1c25486a48289f9b895b080dd2, let me know what do you think

**Review by anfimovdm — APPROVED** [review]

**Review by maccelf — APPROVED** [review]

**Review by Korulag — APPROVED** [review]

**Review by maccelf — APPROVED** [review]

**Pull request merged**